### PR TITLE
Adding 3 test crumbs for requirements

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

Adding 3 test crumbs for requirements.

## Why is the change being made?

This is part of on-going requirement work.

---

## Checklist

- [X] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [X] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [X] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [ ] No [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were altered.
- [X] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: armi/materials/tests/test_materials.py [hunk 1]
```diff
@@ -139,6 +144,60 @@ def test_findMaterial(self):
...
+            :tests: R_ARMI_MAT_ORDER
+        """
+        # let's do a quick test of getting a material from the default namespace
+        setMaterialNamespaceOrder(["armi.materials"])
+        uo2 = materials.resolveMaterialClassByName(
+            # "Hafnium", namespaceOrder=["armi.materials"]
+            "UO2",
+            namespaceOrder=["armi.materials"],
+        )
+        self.assertGreater(uo2().density(500), 0)
```
Comment (albeanth) [inline armi/materials/tests/test_materials.py:173]: are lines 168 through 173 necessary to test the name space?

File: armi/materials/tests/test_materials.py [hunk 2]
```diff
@@ -139,6 +144,60 @@ def test_findMaterial(self):
...
+        # show that adding a name material namespace provides access to new materials
+        testMatIgnoreFake = materials.resolveMaterialClassByName(
+            "TestMaterialIgnoreFake", namespaceOrder=["armi.materials", newMats]
+        )
+        for t in range(200, 600):
+            self.assertEqual(testMatIgnoreFake().density(t), 0)
+            self.assertEqual(testMatIgnoreFake().pseudoDensity(t), 0)
+
+        # show the ordering of the material namespaces works
+        # (The second material namespace would set the density of UO2 to zero.)
```
Comment (albeanth) [inline armi/materials/tests/test_materials.py:192]: I think your second comment would be a good to assertion to show.

File: armi/materials/tests/test_materials.py (hunk 1 above)
Comment (albeanth) [reply armi/materials/tests/test_materials.py:173]: end of the day confusion on my end...

File: armi/physics/neutronics/tests/test_macroXSGenerationInterface.py [hunk 3]
```diff
@@ -13,20 +13,42 @@
...
+            numGroups = 1
+            numGroupsGamma = 1
+
+            def getNuclide(self, nucName, suffix):
+                try:
+                    return nuclideBases.byName.get(nucName, None)
+                except AttributeError:
+                    return None
+
+        i.buildMacros(MockLib(), buildScatterMatrix=False)
```
[the remark this answers is not in the document]
Comment (john-science) [reply armi/physics/neutronics/tests/test_macroXSGenerationInterface.py]: Okay, I significantly beefed up this test.  I generated a completely different mock-nuclide-library.

And now we prove before and after `buildMacros()`, that they were added to the reactor.

Check it out.

File: armi/materials/tests/test_materials.py (hunk 2 above)
Comment (john-science) [reply armi/materials/tests/test_materials.py]: Okay, this and some other stuff was left over from a different version of the tests. I've cleaned a ton of this out.

File: armi/physics/neutronics/tests/test_macroXSGenerationInterface.py [hunk 4]
```diff
@@ -13,20 +13,51 @@
...
+        mockLib.__dict__["_nuclides"] = defaultdict(
+            lambda: mockLib.__dict__["_nuclides"]["CAA"], mockLib.__dict__["_nuclides"]
+        )
+
+        # This is the meat of it: build the macro XS
+        self.assertIsNone(i.macrosLastBuiltAt)
+        i.buildMacros(mockLib, buildScatterMatrix=False)
+        self.assertEqual(i.macrosLastBuiltAt, 0)
+
+        # After: verify there are no macro XS in the reactor
```
Comment (albeanth) [inline armi/physics/neutronics/tests/test_macroXSGenerationInterface.py:60]:
```suggestion
        # After: verify there are macro XS on each block
```

**Review by albeanth — CHANGES_REQUESTED** [review]
just a few minor typos/suggestions in some comments

**Review by albeanth — APPROVED** [review]

**Pull request merged**